### PR TITLE
Update scripts for new semantic segmentation model training

### PR DESCRIPTION
**Pull request by titusgriebel**
@anwai98 This is the current stage of semantic training

### CODE REVIEW

**Review by anwai98 — COMMENTED** [review]
Hi @titusgriebel,

Thanks for the PR. The changes on a higher level looked good to me.

I went ahead and simplified some code, as I couldn't read them at all (eg. the remapping label script) and removed some redundant scripts!

A few quick mentions:
- You need to pull the latest `torch-em` to make sure you have all dataset updates!
- I changed a few scripts here and there. Make sure everything works as expected, as I didn't test them (eg. check if the label mapping is working correctly, if the class weight extraction works as expected, the dataloaders are fine, etc)
- Before we run any training, we need to weight the cross entropy loss for sure. Let's discuss this first thing tomorrow.

Besides that, you can check for dry runs if the training atleast works. And make sure that the class mapping worked (technically and scientifically) correctly (with scientifically, remember our discussion from today ;) )

File: experiments/data/semantic/get_semantic_paths.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            images = images[:]
+            labels = labels[:]
+            # CONIC is provided in an array of shape (C, B, H, W)
+            images = images.transpose(1, 2, 3, 0)  # --> (B, H, W, C)
+            counter = 1
+            for image, label in zip(images, labels):
+                image_path = os.path.join(cached_images, f"{counter:04}.tiff")
+                label_path = os.path.join(cached_labels, f"{counter:04}.tiff")
+                label = remap_conic(label)
+                assert image.shape == (256, 256, 3)
```
[the remark this answers is not in the document]
Comment (titusgriebel) [reply experiments/data/semantic/get_semantic_paths.py:37]: Yes, they are all the same shape. I don't know about CONIC, but the ROI / full image choice was for Lizard.

File: experiments/data/semantic/get_semantic_paths.py
[the remark this answers is not in the document]
Comment (titusgriebel) [reply experiments/data/semantic/get_semantic_paths.py:1]: Yes.

File: experiments/data/semantic/get_semantic_paths.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (anwai98) [reply experiments/data/semantic/get_semantic_paths.py:37]: So in short, CONIC comes with all images of shape (256, 256)?

[the remark this answers is not in the document]
Comment (titusgriebel) [reply experiments/data/semantic/get_semantic_paths.py:37]: Yes, as they provide it in a single array

[the remark this answers is not in the document]
Comment (anwai98) [reply experiments/data/semantic/get_semantic_paths.py:37]: Ah yeah you are right. Thanks!

File: patho_sam/training/util.py [hunk 2]
```diff
@@ -35,3 +81,71 @@ def get_train_val_split(
...
+    # Make sure that the most common class has weight 1.
+    mapped_weights[np.argmax(foreground_class_weights)] = background_weight
+
+    # Round the weights and convert them to integer values.
+    final_weights = np.round(mapped_weights).astype(int)
+
+    # Add background weights in the beginning.
+    final_weights_with_bg = [background_weight, *final_weights]
+
+    return final_weights_with_bg
```
Comment (anwai98) [inline patho_sam/training/util.py:151]: @constantinpape 

from what we discussed to weigh the classes in the loss, I think this is what you meant right?

PS. Our (foreground) class weights are `[0.4702, 0.1797, 0.2229, 0.0159, 0.1113]`, which with this logic now map to `[1, 7, 6, 10, 8]`.

Comment (anwai98) [reply patho_sam/training/util.py:151]: I did a simple relative mapping. Let me know if this does not look like a good balance.

PPS. With class weights, cross-entropy would finally get something like: `[1, 1, 7, 6, 10, 8]`

Comment (constantinpape) [reply patho_sam/training/util.py:139]: Yes, that is what I meant. 10 is the dead cells, correct?

**Review by constantinpape — COMMENTED** [review]
I only skimmed the code, but the class weights look correct to me.

**Comment by anwai98** [comment]
I'll do some clean up later. For now, this is fine and I'll go ahead and merge this!
